### PR TITLE
implement date() and time() in QgsDateTimeEdit to handle NULL values

### PR DESCRIPTION
**Pull request by Gustry**
## Description

implement date() and time() in QgsDateTimeEdit to handle NULL values

Follow up of https://github.com/qgis/QGIS/pull/31855

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [x] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### CODE REVIEW

File: src/gui/editorwidgets/qgsdatetimeedit.h [hunk 1]
```diff
@@ -63,11 +64,23 @@ class GUI_EXPORT QgsDateTimeEdit : public QDateTimeEdit
...
     /**
      * \brief dateTime returns the date time which can eventually be a null date/time
-     * \note  You mustn't call date() or time() because they can't return a NULL value.
+     * \note Before QGIS 3.10, you mustn't call date() or time() because they can't return a NULL value.
      * \note since QDateTimeEdit::dateTime() is not virtual, dateTime must be called for QgsDateTimeEdit
      */
     QDateTime dateTime() const;
 
+    /**
+     * \brief time returns the time which can eventually be a null time.
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsdatetimeedit.h]: eventually != éventuellement

```suggestion
     * \brief time returns the time which can be a null time.
```

Comment (Gustry) [reply src/gui/editorwidgets/qgsdatetimeedit.h]: Ah thanks ! This French "false" friend! @3nids started this class 5 years ago ;-)

Going to fix them ;-)

File: python/gui/auto_generated/editorwidgets/qgsdatetimeedit.sip.in [hunk 2]
```diff
@@ -58,17 +59,31 @@ setDateTime set the date time in the widget and handles null date times.
 
     QDateTime dateTime() const;
 %Docstring
-dateTime returns the date time which can eventually be a null date/time
+dateTime returns the date time which can be a null date/time
```
Comment (nyalldawson) [inline python/gui/auto_generated/editorwidgets/qgsdatetimeedit.sip.in]: This should just be "Returns the...." (Remove the leading "datetime")

Comment (Gustry) [reply python/gui/auto_generated/editorwidgets/qgsdatetimeedit.sip.in]: Done, I have done a review in the file.

**Review by rldhont — APPROVED** [review]

**Pull request merged**